prism_compile.c

```diff
@@ -7232,13 +7242,14 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
                   case PM_INTERPOLATED_REGULAR_EXPRESSION_NODE: {
                     pm_interpolated_regular_expression_node_t *cast = (pm_interpolated_regular_expression_node_t *) scope_node->ast_node;
 
-                    int parts_size = (int)cast->parts.size;
+                    int parts_size = 0;
                     if (parts_size > 0 && !PM_NODE_TYPE_P(cast->parts.nodes[0], PM_STRING_NODE)) {
```

Comment:
Should this be checking the parts size? Looks like the first check here will never be true.

Reply:
Yeah I think this is dead code. We don't need to push the initial empty string anymore because `pm_interpolated_node_compile` handles it.